core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/DashboardResource.scala

```diff
@@ -0,0 +1,238 @@
+package edu.uci.ics.texera.web.resource.dashboard
+
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.auth.SessionUser
+import edu.uci.ics.texera.web.model.jooq.generated.Tables._
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos._
+import edu.uci.ics.texera.web.resource.dashboard.DashboardResource._
+import edu.uci.ics.texera.web.resource.dashboard.user.workflow.WorkflowResource._
+import io.dropwizard.auth.Auth
+import org.jooq.Condition
+import org.jooq.impl.{DSL, SQLDataType}
+import org.jooq.impl.DSL.{groupConcat, noCondition}
+import org.jooq.types.UInteger
+
+import java.sql.Timestamp
+import javax.ws.rs._
+import javax.ws.rs.core.MediaType
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+
+/**
+  * This file handles various requests that need to interact with multiple tables.
+  */
+object DashboardResource {
+  final private lazy val context = SqlServer.createDSLContext()
+  // Add any additional DAOs or objects you need
+
+  case class DashboardClickableFileEntry(
+      resourceType: String, // workflow, project, or file
+      id: UInteger, // wid for workflow, pid for project, fid for file
+      name: String, // name for the ClickableFile
+      description: String, // description for the ClickableFile
+      creation_time: Timestamp, //creation_time for workflow and project, upload_time for file
+      file_path: String, // path of file, set to NULL for workflow and project
+      file_size: UInteger // size of file, set to 0 for workflow and project
+  )
+}
+
+@Produces(Array(MediaType.APPLICATION_JSON))
+@Path("/dashboard")
+class DashboardResource {
+
+  /**
+    * This method performs a full-text search in all resources(workflow, project, file)
+    * that match the specified keywords.
+    *
+    * This method utilizes MySQL Boolean Full-Text Searches
+    * reference: https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html
+    *
+    * @param sessionUser The authenticated user.
+    * @param keywords    The search keywords.
+    * @return A list of DashboardClickableFileEntry that match the search term.
+    */
+  @GET
+  @Path("/searchAll")
```

Comment:
I think we should just name this `search`.

Reply:
done